gcloud-java-storage/src/main/java/com/google/gcloud/spi/DefaultStorageRpc.java

```diff
@@ -465,15 +465,18 @@ public void write(String uploadId, byte[] toWrite, int toWriteOffset, long destO
       GenericUrl url = new GenericUrl(uploadId);
       HttpRequest httpRequest = storage.getRequestFactory().buildPutRequest(url,
           new ByteArrayContent(null, toWrite, toWriteOffset, length));
-      long limit = destOffset + length;
-      StringBuilder range = new StringBuilder("bytes ");
-      range.append(destOffset).append('-').append(limit - 1).append('/');
-      if (last) {
-        range.append(limit);
-      } else {
-        range.append('*');
+      // 0-length requests are valid, don't need content-range.
```

Comment:
I for one am glad that I used this form of `createFile` since it revealed a bug that we were then able to fix :)

I know the file was written to GCS because Files.exists returns true. However it's possible that `createFile` called other methods as well.